zerver/tests/test_message_edit.py

```diff
@@ -1596,3 +1596,62 @@ def test_user_group_mention_restrictions_while_editing(self) -> None:
             },
         )
         self.assert_json_success(result)
+
+        # Test all the cases when can_mention_group is not a named user group.
+        content = "Test mentioning user group @*leadership*"
+        user_group = UserGroup.objects.create(realm=iago.realm)
+        user_group.direct_members.set([othello])
+        user_group.direct_subgroups.set([moderators_system_group])
+        leadership.can_mention_group = user_group
+        leadership.save()
```

Comment:
I think we should aim to use higher-level functions than this in tests wherever possible -- there's no reason we shouldn't be able to just call the `do_` with the object we want in about the same number of lines of code.